manager/manager.go

```diff
@@ -937,12 +943,27 @@ func (m *Manager) becomeLeader(ctx context.Context) {
 			api.EncryptionConfig{AutoLockManagers: m.config.AutoLockManagers},
 			unlockKeys,
 			rootCA,
-			m.config.FIPS))
+			m.config.FIPS,
+			nil,
+			0),
+		)
 
 		if err != nil && err != store.ErrExist {
 			log.G(ctx).WithError(err).Errorf("error creating cluster object")
 		}
-
+		var defaultAddrPool []string
+		for _, p := range m.config.DefaultAddrPool {
+			defaultAddrPool = append(defaultAddrPool, p.String())
+		}
+		// if defaultAddrPool is valid we update cluster object with new value
+		if defaultAddrPool != nil {
+			cluster := store.GetCluster(tx, clusterID)
+			if cluster != nil {
+				cluster.DefaultAddressPool = defaultAddrPool
+				cluster.SubnetSize = uint32(m.config.SubnetSize)
+			}
+			store.UpdateCluster(tx, cluster)
```

Comment:
don't think you need to do a separate update cluster call. You can just update the object returned by defaultClusterObject() before you call store.CreateCluster().

The other issue with this is that this will allow you to change the default address poll post-init, which may not be desired.

I would also suggest adding a unit-test for this very scenario.

Reply:
Anshul, moving attribute to network.go, shall we do as separate PR?

Reply:
I had offline call with Anshul and modified code accordingly. Also we decided we don't need unit test for this change.